store_test.go

```diff
@@ -282,3 +286,29 @@ func idListMapToSyncMap(m map[string]bool) *sync.Map {
 	}
 	return &mm
 }
+
+func getCounter(val *int32) int32 {
+	return atomic.LoadInt32(val)
+}
+
+func incrementCounter(val *int32) {
+	atomic.AddInt32((*int32)(val), 1)
```

Comment:
The type cast shouldn't be necessary here. Looks like we're casting (*int32) to (*int32).